programs/ziti-edge-tunnel/netif_driver/linux/resolvers.c

```diff
@@ -468,49 +502,74 @@ void dns_update_etc_resolv(const char* tun, unsigned int ifindex, const char* ad
     exit(EXIT_FAILURE); \
 } while(0)
 
-#define CHECK_F_EOF(f) do { \
-    if ((f) == EOF) { \
-        ZITI_LOG(ERROR, "EOF received while writing file"); \
-        CLEANUP_ETC_RESOLV(); \
-    } \
-} while (0)
+      struct stat file_stat;
+      if(stat(RESOLV_CONF_FILE, &file_stat) == -1){
+          ZITI_LOG(ERROR, "cannot stat %s: %s", RESOLV_CONF_FILE, strerror(errno));
+          CLEANUP_ETC_RESOLV();
+      }
 
       // Slices everything after the matched line into a buffer,
       // inserts the ziti nameserver line, and flushes the buffer back to the file.
       if (match_start_offset >= 0) {
-          struct stat file_stat;
-          if(stat(RESOLV_CONF_FILE, &file_stat) == -1){
-              ZITI_LOG(ERROR, "cannot stat %s: %s", RESOLV_CONF_FILE, strerror(errno));
-              CLEANUP_ETC_RESOLV();
-          }
 
-          long remaining_size = file_stat.st_size - match_start_offset;
-          _cleanup_(cleanup_bufferp) char* remaining_content = malloc(remaining_size + 1);
+          off_t remaining_size = file_stat.st_size - match_start_offset;
+
+          _cleanup_(cleanup_bufferp) char *remaining_content = (char *)malloc(remaining_size + 1);
           if (remaining_content == NULL) {
-              ZITI_LOG(ERROR, "error allocating %s file buffer: %s", RESOLV_CONF_FILE, strerror(errno));
+              ZITI_LOG(ERROR, "error allocating %s remaining content buffer: %s", RESOLV_CONF_FILE, strerror(errno));
               CLEANUP_ETC_RESOLV();
           }
 
           fseek(file, match_start_offset, SEEK_SET);
-          if (fread(remaining_content, sizeof(char), remaining_size, file) != remaining_size) {
+          if (fread(remaining_content, sizeof(char), (size_t)remaining_size, file) != remaining_size) {
               if (ferror(file) || feof(file)) {
                   ZITI_LOG(ERROR, "Error during file stream operation or EOF received.");
               }
+              cleanup_bufferp(&remaining_content);
+              CLEANUP_ETC_RESOLV();
+          }
+          remaining_content[remaining_size] = '\0';
+
+          _cleanup_(cleanup_bufferp) char *rptr = realloc(replace, (size_t)(replace_size + remaining_size + 1));
```

Comment:
why use a new variable here? I thought the common `realloc` pattern was `p = realloc(p, size);` ?

Reply:
I had some doubts about correct usage here after careful reading of the man page (it has very quirky behavior), so I took inspiration from information shared here:

https://stackoverflow.com/questions/44789295/correct-use-of-realloc

Additionally, there is interplay with the `cleanup()` function guard here I'm being careful to consider so we don't double `free()` in the case where the allocated memory block is moved from it's original location since `realloc()` will `free()` it internally. See #L534-544